amivapi/auth/oauth.py

```diff
@@ -0,0 +1,219 @@
+"""Implement OAuth functionality.
+
+We currently only support the implicit grant type. Client applications
+redirect their users to the /oauth/authorize endpoint of the API. This will
+serve the login page. After correct login or user confirmation (if logged in
+already), the user is referred back to the client application.
+"""
+
+from urllib.parse import urlencode
+
+from eve.methods.post import post_internal
+from flask import (
+    abort,
+    Blueprint,
+    current_app,
+    g,
+    redirect,
+    render_template,
+    request,
+    url_for
+)
+from werkzeug.exceptions import Unauthorized
+
+from amivapi.auth.auth import AdminOnlyAuth, authenticate_token
+from amivapi.utils import register_domain
+from amivapi.check_utils import check_str_nonempty
+
+
+oauth_blueprint = Blueprint('oauth', __name__, template_folder='templates')
+
+
+def append_url_params(url, params):
+    """Add parameters to a url.
+
+    Args:
+        url(str): Existing URL.
+        **params: Parameters to append.
+    """
+    if '?' not in url:
+        url += '?'
+
+    if url[-1] not in ['?', '&']:
+        url += '&'
+
+    return url + urlencode(params)
```

Comment:
We are trying to join two query strings. The function you linked creates a query string from a dictionary. I am already doing that with urlencode from the standard library. I can't see a difference to the werkzeug function, so I don't see why we would replace it.

Reply:
Yes, you are right, I must have not been very awake when reviewing this.
Afaik there is no simpler way of appending query parameters, so this function is fine.

I still think we can simplify this function a little:

As far as I know, multiple `&` are no problems in query strings, i.e. `?test=1` is as valid as `?&test=1` or `?&&test=1`. Therefore the check if the string ends with `?` or `&` can be removed.

I'll quickly commit this since its a very small correction.